lua/cmp/source.lua

```diff
@@ -119,21 +115,15 @@ source.get_entries = function(self, ctx)
       e.exact = e:get_filter_text() == inputs[o] or e:get_word() == inputs[o]
 
       if entry_filter(e, ctx) then
-        table.insert(entries, e)
+        entries[#entries + 1] = e
+        if max_item_count and #entries >= max_item_count then
+          break
+        end
       end
     end
   end
-  self.cache:set({ 'get_entries', tostring(self.revision), ctx.cursor_before_line }, entries)
```

Comment:
I think `get_entries` cache will reduce the CPU cost. Why do you remove this?

Reply:
It leads to big memory issues with tailwind and probably others.

Reply:
I'll see if I can incorporate the cache again in some other way that doesn't include all the memory overhead

Reply:
I believe this solution is actually optimal. When the entries are huge, we filter and break as soon as we have 200 matches.

As soon as we filter down to less than 200 matches, the cache is used and the big entries list no longer needs to be fully processed.

Reply:
I agree that this fix improves the performance.
It also makes sense that the current nvim-cmp is already an incorrect implementation, so there is no negative behavior.

However, I doubt that this approach is the right way.

That being said, I'm not planning any drastic improvements to nvim-cmp, so I agree with efficiency first.